src/orbit-chain-governance/factories/GovernanceChainGovFactory.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-Licens©e-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "../../L2ArbitrumGovernor.sol";
+import "../../ArbitrumTimelock.sol";
+import "../../UpgradeExecutor.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "@openzeppelin/contracts/proxy/transparent/ProxyAdmin.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts-upgradeable/governance/utils/IVotesUpgradeable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+
+/// @notice Parameters for deployment
+struct DeployParams {
+    address _governanceToken; // Address of IVotesUpgradeable token deployed on the governance chain
+    address _govChainUpExec; // Address of governance UpgradeExecutor deployed on the governance chain
+    address _govChainProxyAdmin; // Address of ProxyAdmin deployed on the governance chain
+    uint256 _proposalThreshold; // Number of votes required to submit a proposal
+    uint256 _votingPeriod; // Time period in blocks during which voting for a proposal occurs
+    uint256 _votingDelay; // Delay in blocks after a proposal is submitted before voting begins
+    uint256 _minTimelockDelay; // Delay in seconds after proposal passes before it can be executed
+    uint64 _minPeriodAfterQuorum; // Minimum voting time in blocks after a quorum is reached
+    uint256 _coreQuorumThreshold; // Required quorum for proposal to pass; has 10k denominator
+}
+/// @title Factory that deploys governance chain contracts for cross chain governance
+/// @notice Requires an UpgradeExecutor, a ProxyAdmin, and a governance token
+/// that implements IVotesUpgradeable to already be deployed on the governance chain.
+/// To be executed prior to ParentChainGovFactory on the parent chain.
+
+contract GovernanceChainGovFactory is Ownable {
```

Comment:
Can we add a test that deploys both of these contracts? Maybe we can run a modified verifier script against them too?